openedx/studio-frontend
# feat: generate currentlySupportedLangs.jsx dynamically | FC-0012

## Changes

Generate the list of languages, instead of hardcoded `transifex_langs = "ar,fr,es_419,zh_CN"` in `Makefile`.

Now it generates the file based on the `atlas pull` result.

This pull request is extracted from the deprecation PR below:

 - https://github.com/openedx/studio-frontend/pull/383#discussion_r1363335941

We plan to rebase the other PR once this PR is merged to make it a deprecation-only PR.

## Testing instructions

Run the following command:

```
$ make OPENEDX_ATLAS_PULL=yes pull_translations
$ nvm use 16
$ npm ci
$ npm run build  # Check for build errors if exists
```

A sanity check for the generated `src/i18n/messages/currentlySupportedLangs.jsx` can be useful.

## Refs
This pull request is part of the [FC-0012 project](https://openedx.atlassian.net/l/cp/XGS0iCcQ) which implements the [Translation Infrastructure update OEP-58](https://docs.openedx.org/en/latest/developers/concepts/oep58.html).
